### PR TITLE
replace get_events with has_event in if-else statement

### PR DESCRIPTION
**Pull request by deanlee**
it's faster and more memory efficient.

### CODE REVIEW

**Comment by pd0wm** [comment]
How much cpu does this save?

**Comment by deanlee** [comment]
it is negligible.
just want to avoiding unnecessary memory allocation in a long time looping.

File: selfdrive/controls/controlsd.py [hunk 1]
```diff
@@ -168,37 +169,37 @@ def state_transition(frame, CS, CP, state, events, soft_disable_timer, v_cruise_
...
   elif state == State.enabled:
-    if get_events(events, [ET.USER_DISABLE]):
+    if has_event(events, [ET.USER_DISABLE]):
       state = State.disabled
       AM.add(frame, "disable", enabled)
 
-    elif get_events(events, [ET.IMMEDIATE_DISABLE]):
+    elif has_event(events, [ET.IMMEDIATE_DISABLE]):
       state = State.disabled
       for e in get_events(events, [ET.IMMEDIATE_DISABLE]):
```
Comment (pd0wm) [inline selfdrive/controls/controlsd.py:178]: The same lookup is done twice here. Can we combine them in a smart way? Maybe convert the events into a dict first of event type to list of events of that type.

Comment (pd0wm) [reply selfdrive/controls/controlsd.py:178]: In general there is a lot of repeated code, is it possible to abstract some of it into a function.

Comment (deanlee) [reply selfdrive/controls/controlsd.py:180]: I have added a set to do this. dict will create lists for types everytime we might not need.

**Comment by geohot** [comment]
This adds complexity and 31 lines($1550), and I'm not sure what the gain is.

**Comment by deanlee** [comment]
Oh I See

**Pull request closed without merging**